src/main/java/core/mate/academy/model/Machine.java

```diff
@@ -1,6 +1,8 @@
 package core.mate.academy.model;
 
-public abstract class Machine implements Workable {
+import core.mate.academy.service.MachineProducer;
+
+public abstract class Machine implements Workable, MachineProducer {
```

Comment:
you should create producer classes for each machine TruckProducer ...
```suggestion
public abstract class Machine implements Workable {
```